.github/workflows/milestone-tracking.yml

```diff
@@ -0,0 +1,41 @@
+name: Milestone tracking
+on:
+  pull_request:
+    types: [closed]
+jobs:
+  add_milestone:
+    runs-on: ubuntu-latest
+    if: ${{ github.repository == 'dotnet/roslyn' && github.event.pull_request.merged_at != null && github.event.pull_request.milestone == null && github.event.pull_request.base.ref == 'main' }}
+    steps:
+    - name: Get milestone data
+      env:
+        ORGANIZATION: dotnet
+        REPOSITORY: roslyn-analyzers
+        MILESTONE_NAME: vNext
+      run: |
+        gh api graphql -f query='
+          query($org: String!, $repo: String!, $milestone: String!) {
+            repository(name: $repo, owner: $org) {
+              milestones(query: $milestone, first: 2) {
```

Comment:
why `first: 2` shouldn't there only be one milestone with a matching name?

Reply:
ah, its the first 2 nodes

Reply:
I wanted to use the equivalent of `.Single()` for validation, but couldn't find it. The query appears to be a substring match, and the docs aren't clear about how to be more specific.

Reply:
➡️ Validation is now added